ProjFS.Mac/PrjFSKext/KauthHandler.cpp

```diff
@@ -584,10 +616,20 @@ KEXT_STATIC int HandleVnodeOperation(
                 SizeOrError xattrResult = Vnode_ReadXattr(currentVnode, PrjFSFileXAttrName, &rootXattr, sizeof(rootXattr));
                 if (xattrResult.error == ENOATTR)
                 {
-                    // Only notify if the file is still a placeholder
+                    // Only notify or block writes if the file is still a placeholder
                     goto CleanupAndReturn;
                 }
                 
+                if (providerStatus == Provider_IsOffline)
+                {
+                    assert(!shouldMessageProvider);
+                    // Deny write access to placeholders in an offline root
+                    kauthResult = KAUTH_RESULT_DENY;
```

Comment:
If we got to this line we don't have the PrjFSFileXAttrName attribute and are a full file correct?

Reply:
No, we *do* have the attribute (as `xattrResult.error != ENOATTR`) and therefore have a placeholder file, so with the provider offline, we want to deny modifications.